### PR TITLE
LOOPP Keystore 

### PR DESCRIPTION
**Pull request by george-dorin**
Add the necessary GRPC components to start a LOOPP Keystore

### CODE REVIEW

File: pkg/loop/internal/pb/keystore/keystore.proto [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  rpc Sign(SignRequest) returns (SignResponse);
+  rpc SignBatch(SignBatchRequest)returns (SignBatchResponse);
+  rpc Verify(VerifyRequest)returns (VerifyResponse);
+  rpc VerifyBatch(VerifyBatchRequest)returns (VerifyBatchResponse);
+
+  rpc List(ListRequest)returns (ListResponse);
+
+  rpc RunUDF(RunUDFRequest)returns (RunUDFResponse);
+
+  //
```
Comment (jmank88) [inline pkg/loop/internal/pb/keystore/keystore.proto:17]:
```suggestion
```

File: pkg/loop/internal/pb/keystore/keystore.proto [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+message ListRequest {
+  repeated string tags = 1;
+}
+
+message ListResponse {
+  repeated bytes keyIDs = 1;
+}
+
+message RunUDFRequest {
+  string udfName = 1;
```
Comment (jmank88) [inline pkg/loop/internal/pb/keystore/keystore.proto:74]: Is the `udf` prefix adding value here?
```suggestion
  string name = 1;
```
Or would it be more clear as `id`?
```suggestion
  string id = 1;
```

File: pkg/loop/internal/pb/keystore/keystore.proto [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  rpc List(ListRequest)returns (ListResponse);
+
+  rpc RunUDF(RunUDFRequest)returns (RunUDFResponse);
+
+  //
+  rpc Import(ImportRequest)returns(ImportResponse);
+  rpc Export(ExportRequest)returns(ExportResponse);
+
+  rpc Create(CreateRequest)returns(CreateResponse);
+  rpc Delete(DeleteRequest)returns(DeleteResponse);
```
Comment (cedric-cordenier) [inline pkg/loop/internal/pb/keystore/keystore.proto:22]: Create and Delete what? a key? a keystore?

File: pkg/loop/internal/keystore/types.go [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package keystore
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+// Methods This interface contains all the functionalities of the GRPC layer of the LOOPP keystore
+type Methods interface {
```
Comment (cedric-cordenier) [inline pkg/loop/internal/keystore/types.go:10]: This isn't going to be super helpful as feedback but Methods is a confusing name IMO -- what are you describing here? It sounds like it's a public interface of some kind, so maybe keystore.Public would be better?

File: pkg/loop/internal/pb/keystore/keystore.proto (hunk 1 above)
Comment (george-dorin) [reply pkg/loop/internal/pb/keystore/keystore.proto]: Removed in [`9bc2713` (#837)](https://github.com/smartcontractkit/chainlink-common/pull/837/commits/9bc2713bc46d6518a758830828454e7077039dbe)

File: pkg/types/keystore/types.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+type Keystore interface {
+	Sign(ctx context.Context, keyID []byte, data []byte) ([]byte, error)
+	SignBatch(ctx context.Context, keyID []byte, data [][]byte) ([][]byte, error)
+	Verify(ctx context.Context, keyID []byte, data []byte) (bool, error)
+	VerifyBatch(ctx context.Context, keyID []byte, data [][]byte) ([]bool, error)
+
+	List(ctx context.Context, tags []string) ([][]byte, error)
+
+	RunUDF(ctx context.Context, udfName string, keyID []byte, data []byte) ([]byte, error)
```
[the remark this answers is not in the document]
Comment (george-dorin) [reply pkg/types/keystore/types.go]: Added in [`9bc2713` (#837)](https://github.com/smartcontractkit/chainlink-common/pull/837/commits/9bc2713bc46d6518a758830828454e7077039dbe)

File: pkg/loop/internal/pb/keystore/keystore.proto (hunk 2 above)
Comment (george-dorin) [reply pkg/loop/internal/pb/keystore/keystore.proto]: Renamed to `name` in [`9bc2713` (#837)](https://github.com/smartcontractkit/chainlink-common/pull/837/commits/9bc2713bc46d6518a758830828454e7077039dbe)

File: pkg/loop/internal/pb/keystore/keystore.proto (hunk 3 above)
Comment (george-dorin) [reply pkg/loop/internal/pb/keystore/keystore.proto]: Renamed the methods in  [`9bc2713` (#837)](https://github.com/smartcontractkit/chainlink-common/pull/837/commits/9bc2713bc46d6518a758830828454e7077039dbe)

File: pkg/loop/internal/keystore/types.go (hunk 4 above)
Comment (george-dorin) [reply pkg/loop/internal/keystore/types.go]: Renamed the interface to illustrate that this is only the GRPC service [`9bc2713` (#837)](https://github.com/smartcontractkit/chainlink-common/pull/837/commits/9bc2713bc46d6518a758830828454e7077039dbe)

**Review by jmank88 — APPROVED** [review]

**Pull request merged**